### DISCUSSION TITLE
ZSH-Function with bat as Syntax highlighter and xclip

### DISCUSSION THREAD

**Discussion by vansoest** [TheR1D/shell_gpt#200] [Show and tell]
Maybe useful for someone. I have implemented this small function for the zsh shell to have fancy syntax code highlighting and auto copying the output to the xorg clipboard with xclip.

```
function sgpt() {
  command sgpt "$@" | tee >(xclip) | bat -l log --no-pager --plain -l C++
}
```

**Comment by moritztim** [comment]
https://gist.github.com/moritz-t-w/f737e41f23d529cce9fb38fa2dfc3180
This version has a `--lang | -l` param for the language. It's used for highlighting and as an instruction to gpt

**Reply by moritztim** [reply]
It doesn't tell sgpt the language tho, you still need to say that in the prompt

**Reply by moritztim** [reply]
```sh
    if [[ ! " $@ " =~ " --code " ]]; then
		command sgpt "${args[@]}" | glow # lang isn't used here but the check above is still necesary to remove the param
    else
		command sgpt "${args[@]}" | tee >(xclip) | bat -l log --no-pager --plain -l "${lang}"
    fi
}
```
With this at the end it only highlights code if the `--code` role is used and renders it as markdown instead. #333

**Reply by moritztim** [reply]
I tried to include the language but that's a headache
